packages/lib/src/components/PayByBankUS/PayByBankUS.test.tsx

```diff
@@ -0,0 +1,39 @@
+import { render, screen } from '@testing-library/preact';
+import PayByBankUS from './PayByBankUS';
+
+describe('PayByBank US', () => {
+    test('should render payment description by default', async () => {
+        const pbb = new PayByBankUS(global.core, {
+            i18n: global.i18n,
+            loadingContext: 'test',
+            modules: { resources: global.resources }
+        });
+
+        render(pbb.render());
+        expect(await screen.findByText(/Use Pay by Bank to pay/i)).toBeTruthy();
+        expect(await screen.findByText(/By connecting your bank account/i)).toBeTruthy();
+    });
+
+    test('should render redirect button by default', async () => {
+        const pbb = new PayByBankUS(global.core, {
+            i18n: global.i18n,
+            loadingContext: 'test',
+            modules: { resources: global.resources }
+        });
+
+        render(pbb.render());
+        expect(await screen.findByRole('button')).toHaveTextContent('Continue to');
+    });
+
+    test('should not render pay button if showPayButton is false', () => {
+        const pbb = new PayByBankUS(global.core, {
+            i18n: global.i18n,
+            loadingContext: 'test',
+            modules: { resources: global.resources },
+            showPayButton: false
+        });
+
+        render(pbb.render());
+        expect(screen.queryByRole('button')).not.toBeInTheDocument();
+    });
+});
```

Comment:
Can you add a test case here just to make sure that the 'submit' flow is triggered once the payment button is pressed (or .submit is called) ?

We are kinda used to test mainly the UI look, but we aren't actually testing the payment part which is very important.

I believe that you can just validate that the `onSubmit` callback is called, then we are sure that the payment flow works.

Reply:
Fixed.